pkg/appgateserver/server.go

```diff
@@ -303,4 +305,25 @@ func (app *appGateServer) ServePprof(ctx context.Context, addr string) error {
 	return nil
 }
 
+// extractServiceId extracts the serviceId from the request path and returns it
+// along with the new request path that is stripped of the serviceId.
+func extractServiceId(urlPath string) (newUrlPath string, serviceId string) {
+	// Extract the serviceId from the request path.
+	serviceId = strings.Split(urlPath, "/")[1]
+
+	// Remove the serviceId from the request path which is specific AppGateServer.
+	// The remaining path is the path of the request that will be serialized and
+	// sent to sent to the supplier within a RelayRequest. For example:
+	// * If a request to the Backend service has to ba made with the path "/backend/relay"
+	// * The client has to send the request to the AppGateServer with the path
+	//   "/serviceId/backend/relay"
+	// * The AppGateServer will remove the serviceId from the path and send the request
+	//   and serialize the request to send to the supplier with the path "/backend/relay"
+	//
+	// This is specific to the AppGateServer and other gateways may have different
+	// means of identifying the service that the request is for.
```

Comment:
```suggestion
	// * The AppGateServer will remove the serviceId from the path, serialize the request,
	//   and send it to the supplier with the path: "/backend/relay"
	//
	// This is specific logic to how the AppGateServer functions. Other gateways may
	// have different means or approaches of identifying the service that the request is for (e.g. POST data).
```

Reply:
Thank you for helping cleaning this giberish :sweat_smile: